aviary/examples/external_subsystems/engine_NPSS/table_engine_builder.py

```diff
@@ -0,0 +1,186 @@
+        for i in range(engine_data.shape[0]):
+            # find the index of the first instance where throttle=1.0
+            index = np.where((engine_data[:, 0] == engine_data[i, 0]) & (
+                engine_data[:, 1] == engine_data[i, 1]) & (engine_data[:, 2] == 1.0))[0][0]
+            thrust_max[i] = engine_data[index, 3]
+
+        print(Dynamic.Mission.THRUST, '--------------------------------------')
+
+        # add inputs and outputs to interpolator
+        engine.add_input(Dynamic.Mission.MACH,
+                         engine_data[:, 0],
+                         units='unitless',
+                         desc='Current flight Mach number')
+        engine.add_input(Dynamic.Mission.ALTITUDE,
+                         engine_data[:, 1],
+                         units='ft',
+                         desc='Current flight altitude')
+        engine.add_input(Dynamic.Mission.THROTTLE,
+                         engine_data[:, 2],
+                         units='unitless',
+                         desc='Current engine throttle')
+        engine.add_output(Dynamic.Mission.THRUST,
+                          engine_data[:, 3],
+                          units='lbf',
+                          desc='Current net thrust produced')
+        engine.add_output(Dynamic.Mission.THRUST_MAX,
+                          thrust_max,
+                          units='lbf',
+                          desc='Max net thrust produced')
+        engine.add_output(Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE,
+                          -engine_data[:, 4],
+                          units='lbm/s',
+                          desc='Current fuel flow rate ')
+        engine.add_output(Dynamic.Mission.ELECTRIC_POWER,
+                          zeros_array,
+                          units='kW',
+                          desc='Current electric energy rate')
+        engine.add_output(Dynamic.Mission.NOX_RATE,
+                          zeros_array,
+                          units='lb/h',
+                          desc='Current NOx emission rate')
+        engine.add_output(Dynamic.Mission.TEMPERATURE_ENGINE_T4,
+                          zeros_array,
+                          units='degR',
+                          desc='Current turbine exit temperature')
+        return engine
+
+    def get_bus_variables(self):
+      # Transfer training data from pre-mission to mission
+        return vars_to_connect
+
+    def get_controls(self, phase_name):
+        '''
+        Builds dictionary of controls for engine off-design.
+
+        Returns
+        -------
+        controls : dict
+            Dictionary with keys that are names of variables to be controlled and the
+            values are dictionaries with the keys `units`, `upper`, and `lower` which states the units of the
+            variable to be controlled.
+        '''
+
+        # TODO: we will need to handle this throttle_dict differently if the
```

Comment:
I think you can remove the electric shaft power stuff from this example, since it is commented out.

Reply:
removed, and uploaded